### ISSUE TITLE
PushKit crashes app on receipt of push

### ISSUE THREAD

**Issue by ghenry** [flutter-webrtc/callkeep#28]
Hi all,

Existing push that works with `flutter_voip_kit` (has updated payload for callkeep):

`curl -v -d '{"aps":{"alert":{"callkeep": {"title":"Incoming call","number":"01224900123"},"badge": {"badge": "0"}}}}' -H "apns-push-type: voip" -H "apns-expiration: 0" -H "apns-priority: 0" -H "apns-topic: blah.voip" --http2 --cert ./cert.pem https://api.sandbox.push.apple.com/3/device/xxx`

app instantly dies.

**Comment by ghenry** [comment]
Helps if I run it on Xcode for the `NSLog` statements etc.:

```
2020-11-11 15:50:23.524499+0000 Runner[605:90134] didReceiveIncomingPushWithPayload payload = PKPushTypeVoIP
2020-11-11 15:50:23.524817+0000 Runner[605:90134] [CallKeep][reportNewIncomingCall] uuidString = 1e38e7d8-5b99-4f1a-b089-bef60f4e8227
2020-11-11 15:50:23.525418+0000 Runner[605:90134] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[CXHandle initWithType:value:]: parameter 'value' cannot be nil'
*** First throw call stack:
(0x180475114 0x193c9bcb4 0x1803782cc 0x19d51a048 0x102e8d750 0x102e8b098 0x1a73395e8 0x103143db8 0x1031455fc 0x103153a6c 0x1803f4c30 0x1803ef0e8 0x1803ee200 0x1964e9598 0x182cb4004 0x182cb95d8 0x10011fc20 0x1800cd598)
libc++abi.dylib: terminating with uncaught exception of type NSException
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[CXHandle initWithType:value:]: parameter 'value' cannot be nil'
terminating with uncaught exception of type NSException
```

**Comment by ghenry** [comment]
Any ideas? Related to #29 ?

**Comment by ghenry** [comment]
Looks like this is an issue with my JSON payload in the push as there needs to be a `number` key. See attached. We also need to override the `hello` string :)

```
2020-12-03 14:45:35.835860+0000 Runner[418:31825] didReceiveIncomingPushWithPayload payload = PKPushTypeVoIP
2020-12-03 14:45:35.836110+0000 Runner[418:31825] [CallKeep][reportNewIncomingCall] uuidString = a662a67f-aecf-4672-a59c-9677a5b4d818
2020-12-03 14:45:35.837082+0000 Runner[418:31825] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[CXHandle initWithType:value:]: parameter 'value' cannot be nil'
*** First throw call stack:
(0x1a55499d4 0x1b8efab54 0x1a544c4c0 0x1c2b5a038 0x1038b1750 0x1038af098 0x1ccb515e8 0x103b17ce4 0x103b19528 0x103b27994 0x1a54c95d0 0x1a54c3a78 0x1a54c2b90 0x1bb7e5598 0x1a7dac638 0x1a7db1bb8 0x100b1e7c8 0x1a51a1588)
libc++abi.dylib: terminating with uncaught exception of type NSException
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[CXHandle initWithType:value:]: parameter 'value' cannot be nil'
terminating with uncaught exception of type NSException
(lldb) ```

<img width="1440" alt="Screenshot 2020-12-03 at 14 56 17" src="https://user-images.githubusercontent.com/159608/101046005-9cec1f80-3578-11eb-967b-48249706dc5e.png">

**Comment by ghenry** [comment]
Helps if I read the payload of my initial issue report. Re-checking.

**Comment by ghenry** [comment]
Solved. Correct PUSH payload for background notifications (and VoIP ones) is like so:

https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/pushing_background_updates_to_your_app

```'{"aps":{"content-available":1},"callkeep":{"number":"01224900123","title":"SureVoIP Office"}}```

**Comment by pedrospecter** [comment]
Thank you for this 🙏

**Comment by md-riaz** [comment]
'{"aps":{"content-available":1},"callkeep":{"number":"01224900123","title":"SureVoIP Office"}}'
is this for both android and ios push notification?
